src/docfx/build/xref/XrefMap.cs

```diff
@@ -70,8 +70,10 @@ public XrefMapModel ToXrefMapModel(Context context)
             var displayPropertyValue = spec.GetXrefPropertyValue(displayPropertyName);
 
             // fallback order:
-            // xrefSpec.displayPropertyName -> xrefSpec.name -> uid
-            var display = !string.IsNullOrEmpty(displayPropertyValue) ? displayPropertyValue : (!string.IsNullOrEmpty(name) ? name : uid);
+            // text -> xrefSpec.displayPropertyName -> xrefSpec.name -> uid
+            var display = !string.IsNullOrEmpty(text)
```

Comment:
can the fallback be handled in `DependencyResolver` ? XrefMap only need to return `xrefSpec`

Reply:
sounds good, then we do not need to pass along `text`

Reply:
We can not calculate display in `DependencyResolver`, because we have `t_recursionDetector` within `XrefMap.Resolve`, and we need to load the lazy value by `var displayPropertyValue = spec.GetXrefPropertyValue(displayPropertyName);` within `XrefMap.Resolve` to be able to detect the cyclic reference.
If we do it in `DependencyResolver`, it will end up into `StackOverflow`.
We can clean up this when we remove the handling of cyclic reference.

Reply:
cool, #3916